doc/man/man3/NewProjectGui.3
.TH "NewProjectGui" 3 "Fri Jun 12 2015" "Version 0.0.1" "RIDE" \" -*- nroff -*-
.ad l
.nh
.SH NAME
NewProjectGui \- 
.SH SYNOPSIS
.br
.PP
.PP
\fC#include <NewProjectGui\&.h>\fP
.PP
Inherits QWidget\&.
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "\fBNewProjectGui\fP (QWidget *parent=0)"
.br
.ti -1c
.RI "void \fBinitBtns\fP ()"
.br
.ti -1c
.RI "void \fBswapNextPage\fP ()"
.br
.ti -1c
.RI "void \fBswapBackPage\fP ()"
.br
.ti -1c
.RI "void \fBloadPage_1\fP ()"
.br
.ti -1c
.RI "void \fBunloadPage_1\fP ()"
.br
.ti -1c
.RI "void \fBloadPage_2\fP ()"
.br
.ti -1c
.RI "void \fBunloadPage_2\fP ()"
.br
.ti -1c
.RI "void \fBloadPage_3\fP ()"
.br
.ti -1c
.RI "void \fBunloadPage_3\fP ()"
.br
.ti -1c
.RI "void \fBloadPage_4\fP ()"
.br
.ti -1c
.RI "void \fBunloadPage_4\fP ()"
.br
.ti -1c
.RI "void \fBgenRideProjDepends\fP (QProcess *process, QString *projectRootStrPtr)"
.br
.ti -1c
.RI "void \fBsetAbsPathToProjBuildFileStrPtr\fP (QString *\fBabsPathToProjBuildFileStrPtr\fP)"
.br
.ti -1c
.RI "QString * \fBgetAbsPathToProjBuildFileStrPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetAbsPathToProjRunFileStrPtr\fP (QString *\fBabsPathToProjRunFileStrPtr\fP)"
.br
.ti -1c
.RI "QString * \fBgetAbsPathToProjRunFileStrPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetAbsPathToProjLaunchFileStrPtr\fP (QString *\fBabsPathToProjLaunchFileStrPtr\fP)"
.br
.ti -1c
.RI "QString * \fBgetAbsPathToProjLaunchFileStrPtr\fP ()"
.br
.ti -1c
.RI "QString * \fBtoString\fP ()"
.br
.ti -1c
.RI "\fB~NewProjectGui\fP ()"
.br
.in -1c
.SS "Private Types"

.in +1c
.ti -1c
.RI "enum \fBPage\fP { \fBPAGE_ONE\fP, \fBPAGE_TWO\fP, \fBPAGE_THREE\fP, \fBPAGE_FOUR\fP }"
.br
.in -1c
.SS "Private Slots"

.in +1c
.ti -1c
.RI "void \fBhandleBackBtnSlot\fP ()"
.br
.ti -1c
.RI "void \fBhandleNextBtnSlot\fP ()"
.br
.ti -1c
.RI "void \fBhandleFinishBtnSlot\fP ()"
.br
.ti -1c
.RI "void \fBhandleHelpBntSlot\fP ()"
.br
.ti -1c
.RI "void \fBhandleCancelBtnSlot\fP ()"
.br
.in -1c
.SS "Private Attributes"

.in +1c
.ti -1c
.RI "\fBPage\fP \fBcurrentPage\fP"
.br
.ti -1c
.RI "QPushButton * \fBbackBtn\fP"
.br
.ti -1c
.RI "QPushButton * \fBnextBtn\fP"
.br
.ti -1c
.RI "QPushButton * \fBfinishBtn\fP"
.br
.ti -1c
.RI "QPushButton * \fBhelpBtn\fP"
.br
.ti -1c
.RI "QPushButton * \fBcancelBtn\fP"
.br
.ti -1c
.RI "QGridLayout * \fBouterLayout\fP"
.br
.ti -1c
.RI "QHBoxLayout * \fBbuttonLayout\fP"
.br
.ti -1c
.RI "\fBNewProjectPage_1\fP * \fBnewProjectPage_1Ptr\fP"
.br
.ti -1c
.RI "\fBNewProjectPage_2\fP * \fBnewProjectPage_2Ptr\fP"
.br
.ti -1c
.RI "\fBNewProjectPage_3\fP * \fBnewProjectPage_3Ptr\fP"
.br
.ti -1c
.RI "\fBNewProjectPage_4\fP * \fBnewProjectPage_4Ptr\fP"
.br
.ti -1c
.RI "QString * \fBabsPathToProjBuildFileStrPtr\fP"
.br
.ti -1c
.RI "QString * \fBabsPathToProjRunFileStrPtr\fP"
.br
.ti -1c
.RI "QString * \fBabsPathToProjLaunchFileStrPtr\fP"
.br
.ti -1c
.RI "QString * \fBabsPathToProjCreateFileStrPtr\fP"
.br
.in -1c
.SH "Detailed Description"
.PP 
Definition at line 32 of file NewProjectGui\&.h\&.
.SH "Member Enumeration Documentation"
.PP 
.SS "enum \fBNewProjectGui::Page\fP\fC [private]\fP"

.PP
\fBEnumerator\fP
.in +1c
.TP
\fB\fIPAGE_ONE \fP\fP
.TP
\fB\fIPAGE_TWO \fP\fP
.TP
\fB\fIPAGE_THREE \fP\fP
.TP
\fB\fIPAGE_FOUR \fP\fP
.PP
Definition at line 37 of file NewProjectGui\&.h\&.
.SH "Constructor & Destructor Documentation"
.PP 
.SS "NewProjectGui::NewProjectGui (QWidget *parent = \fC0\fP)"

.PP
Definition at line 14 of file NewProjectGui\&.cpp\&.
.SS "NewProjectGui::~NewProjectGui ()"

.PP
Definition at line 426 of file NewProjectGui\&.cpp\&.
.SH "Member Function Documentation"
.PP 
.SS "void NewProjectGui::genRideProjDepends (QProcess *process, QString *projectRootStrPtr)"

.PP
Definition at line 268 of file NewProjectGui\&.cpp\&.
.SS "QString * NewProjectGui::getAbsPathToProjBuildFileStrPtr ()"

.PP
Definition at line 375 of file NewProjectGui\&.cpp\&.
.SS "QString * NewProjectGui::getAbsPathToProjLaunchFileStrPtr ()"

.PP
Definition at line 399 of file NewProjectGui\&.cpp\&.
.SS "QString * NewProjectGui::getAbsPathToProjRunFileStrPtr ()"

.PP
Definition at line 387 of file NewProjectGui\&.cpp\&.
.SS "void NewProjectGui::handleBackBtnSlot ()\fC [private]\fP, \fC [slot]\fP"

.PP
Definition at line 41 of file NewProjectGui\&.cpp\&.
.SS "void NewProjectGui::handleCancelBtnSlot ()\fC [private]\fP, \fC [slot]\fP"

.PP
Definition at line 116 of file NewProjectGui\&.cpp\&.
.SS "void NewProjectGui::handleFinishBtnSlot ()\fC [private]\fP, \fC [slot]\fP"

.PP
Definition at line 54 of file NewProjectGui\&.cpp\&.
.SS "void NewProjectGui::handleHelpBntSlot ()\fC [private]\fP, \fC [slot]\fP"

.PP
Definition at line 110 of file NewProjectGui\&.cpp\&.
.SS "void NewProjectGui::handleNextBtnSlot ()\fC [private]\fP, \fC [slot]\fP"

.PP
Definition at line 47 of file NewProjectGui\&.cpp\&.
.SS "void NewProjectGui::initBtns ()"

.PP
Definition at line 122 of file NewProjectGui\&.cpp\&.
.SS "void NewProjectGui::loadPage_1 ()"

.PP
Definition at line 204 of file NewProjectGui\&.cpp\&.
.SS "void NewProjectGui::loadPage_2 ()"

.PP
Definition at line 220 of file NewProjectGui\&.cpp\&.
.SS "void NewProjectGui::loadPage_3 ()"

.PP
Definition at line 236 of file NewProjectGui\&.cpp\&.
.SS "void NewProjectGui::loadPage_4 ()"

.PP
Definition at line 252 of file NewProjectGui\&.cpp\&.
.SS "void NewProjectGui::setAbsPathToProjBuildFileStrPtr (QString *absPathToProjBuildFileStrPtr)"

.PP
Definition at line 369 of file NewProjectGui\&.cpp\&.
.SS "void NewProjectGui::setAbsPathToProjLaunchFileStrPtr (QString *absPathToProjLaunchFileStrPtr)"

.PP
Definition at line 393 of file NewProjectGui\&.cpp\&.
.SS "void NewProjectGui::setAbsPathToProjRunFileStrPtr (QString *absPathToProjRunFileStrPtr)"

.PP
Definition at line 381 of file NewProjectGui\&.cpp\&.
.SS "void NewProjectGui::swapBackPage ()"

.PP
Definition at line 150 of file NewProjectGui\&.cpp\&.
.SS "void NewProjectGui::swapNextPage ()"

.PP
Definition at line 177 of file NewProjectGui\&.cpp\&.
.SS "QString * NewProjectGui::toString ()"

.PP
Definition at line 405 of file NewProjectGui\&.cpp\&.
.SS "void NewProjectGui::unloadPage_1 ()"

.PP
Definition at line 212 of file NewProjectGui\&.cpp\&.
.SS "void NewProjectGui::unloadPage_2 ()"

.PP
Definition at line 228 of file NewProjectGui\&.cpp\&.
.SS "void NewProjectGui::unloadPage_3 ()"

.PP
Definition at line 244 of file NewProjectGui\&.cpp\&.
.SS "void NewProjectGui::unloadPage_4 ()"

.PP
Definition at line 260 of file NewProjectGui\&.cpp\&.
.SH "Member Data Documentation"
.PP 
.SS "QString* NewProjectGui::absPathToProjBuildFileStrPtr\fC [private]\fP"

.PP
Definition at line 61 of file NewProjectGui\&.h\&.
.SS "QString* NewProjectGui::absPathToProjCreateFileStrPtr\fC [private]\fP"

.PP
Definition at line 64 of file NewProjectGui\&.h\&.
.SS "QString* NewProjectGui::absPathToProjLaunchFileStrPtr\fC [private]\fP"

.PP
Definition at line 63 of file NewProjectGui\&.h\&.
.SS "QString* NewProjectGui::absPathToProjRunFileStrPtr\fC [private]\fP"

.PP
Definition at line 62 of file NewProjectGui\&.h\&.
.SS "QPushButton* NewProjectGui::backBtn\fC [private]\fP"

.PP
Definition at line 47 of file NewProjectGui\&.h\&.
.SS "QHBoxLayout* NewProjectGui::buttonLayout\fC [private]\fP"

.PP
Definition at line 54 of file NewProjectGui\&.h\&.
.SS "QPushButton* NewProjectGui::cancelBtn\fC [private]\fP"

.PP
Definition at line 51 of file NewProjectGui\&.h\&.
.SS "\fBPage\fP NewProjectGui::currentPage\fC [private]\fP"

.PP
Definition at line 45 of file NewProjectGui\&.h\&.
.SS "QPushButton* NewProjectGui::finishBtn\fC [private]\fP"

.PP
Definition at line 49 of file NewProjectGui\&.h\&.
.SS "QPushButton* NewProjectGui::helpBtn\fC [private]\fP"

.PP
Definition at line 50 of file NewProjectGui\&.h\&.
.SS "\fBNewProjectPage_1\fP* NewProjectGui::newProjectPage_1Ptr\fC [private]\fP"

.PP
Definition at line 56 of file NewProjectGui\&.h\&.
.SS "\fBNewProjectPage_2\fP* NewProjectGui::newProjectPage_2Ptr\fC [private]\fP"

.PP
Definition at line 57 of file NewProjectGui\&.h\&.
.SS "\fBNewProjectPage_3\fP* NewProjectGui::newProjectPage_3Ptr\fC [private]\fP"

.PP
Definition at line 58 of file NewProjectGui\&.h\&.
.SS "\fBNewProjectPage_4\fP* NewProjectGui::newProjectPage_4Ptr\fC [private]\fP"

.PP
Definition at line 59 of file NewProjectGui\&.h\&.
.SS "QPushButton* NewProjectGui::nextBtn\fC [private]\fP"

.PP
Definition at line 48 of file NewProjectGui\&.h\&.
.SS "QGridLayout* NewProjectGui::outerLayout\fC [private]\fP"

.PP
Definition at line 53 of file NewProjectGui\&.h\&.

.SH "Author"
.PP 
Generated automatically by Doxygen for RIDE from the source code\&.


src/NewProjectGui.cpp
#include "NewProjectGui.h"

/*
 * #!/bin/bash
 * echo 'catkin_make ' $1 $2
 * 
 * ./test.bash --pkg test_node
 * > catkin_make --pkg test_node
 * 
 * ./test.bash
 * > catkin_make
 */

NewProjectGui::NewProjectGui(QWidget* parent) : QWidget(parent)
{
    this->setWindowIcon(QIcon("/home/james/NetBeansProjects/ride/images/project2.jpg") );
    this->setWindowTitle("Ride");
    
    outerLayout = new QGridLayout(this);
    newProjectPage_1Ptr = new NewProjectPage_1();
    newProjectPage_2Ptr = new NewProjectPage_2();
    newProjectPage_3Ptr = new NewProjectPage_3();
    newProjectPage_4Ptr = new NewProjectPage_4();
    
    currentPage = PAGE_ONE;
    
    initBtns();
    
    this->setLayout(outerLayout);
    /*
    //--------------------------------------------------------------------------
    QFile file("/home/james/NetBeansProjects/Hazel/src/qss/Dark.css");
    file.open(QFile::ReadOnly);
    QString styleSheet = QLatin1String(file.readAll());
    backBtn->setStyleSheet(styleSheet);
    //--------------------------------------------------------------------------
    */
}


void NewProjectGui::handleBackBtnSlot()
{
    swapBackPage();
}


void NewProjectGui::handleNextBtnSlot()
{
    
    swapNextPage();
}


void NewProjectGui::handleFinishBtnSlot()
{
    cout << "\"Finish\" button selected" << endl;
    newProjectPage_1Ptr->triggerMutators();
    cout << "successfully triggered p1" << endl;
    newProjectPage_2Ptr->triggerMutators();
    cout << "successfully triggered p2" << endl;
    newProjectPage_3Ptr->triggerMutators();
    cout << "successfully triggered p3" << endl;
    //newProjectPage_4Ptr updates automatically
    cout << toString()->toStdString() << endl;
    //- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

    QProcess* process = new QProcess();
    QDir dir; dir.setPath(*newProjectPage_2Ptr->getProjectLocStrPtr() );
    process->setWorkingDirectory(dir.absolutePath() );
    QDir::setCurrent(dir.absolutePath() );

    QStringList* args = new QStringList();
    
    args->push_back("--rosdistro");
    args->push_back(*newProjectPage_1Ptr->getRosVersionStrPtr() );
    args->push_back(*newProjectPage_2Ptr->getProjectNameStrPtr() );
    for(size_t i = 0; i < newProjectPage_4Ptr->getDependsEnteredStrList()->size(); i++)
    {
        args->push_back(newProjectPage_4Ptr->getDependsEnteredStrList()->at(i) );
    }

    //args->push_front("indigo");
    //args->push_front("--rosdistro");

    QStringList tmp0; tmp0.push_back("./../devel/setup.bash");
    
    process->setWorkingDirectory(dir.absolutePath() );
    QDir::setCurrent(dir.absolutePath() );
    process->execute("source", tmp0);
    cout << "exe? " << process->startDetached("catkin_create_pkg", *args) << endl;

    
    process->waitForFinished();
    QString output(process->readAllStandardOutput());
    cout << output.toStdString() << endl;
    cout << "\nfinished create!-------------------------" << endl;
    //- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
    FileTreeGui::setProjectRootAbsPathStrPtr(new QString(*newProjectPage_2Ptr->getProjectLocStrPtr()
                                                       + "/"
                                                       + *newProjectPage_2Ptr->getProjectNameStrPtr() ));
    
    
    QString tmp5(*newProjectPage_2Ptr->getProjectLocStrPtr() + "/" + *newProjectPage_2Ptr->getProjectNameStrPtr() );
    genRideProjDepends(process, new QString(tmp5) );
    
    FileTreeGui::refresh();
}


void NewProjectGui::handleHelpBntSlot()
{
    ;
}


void NewProjectGui::handleCancelBtnSlot()
{
    this->close();
}


void NewProjectGui::initBtns()
{
    backBtn = new QPushButton("< Back", this);
    backBtn->setEnabled(false);
    nextBtn = new QPushButton("Next >", this);
    nextBtn->autoDefault();
    finishBtn = new QPushButton("Finish", this);
    finishBtn->setEnabled(false);
    helpBtn = new QPushButton("Help", this);
    cancelBtn = new QPushButton("Cancel", this);
    
    buttonLayout = new QHBoxLayout();
    buttonLayout->addWidget(backBtn);
    buttonLayout->addWidget(nextBtn);
    buttonLayout->addWidget(finishBtn);
    buttonLayout->addWidget(helpBtn);
    buttonLayout->addWidget(cancelBtn);
    
    connect(backBtn, SIGNAL(released() ), this, SLOT(handleBackBtnSlot() ) );
    connect(nextBtn, SIGNAL(released() ), this, SLOT(handleNextBtnSlot() ) );
    connect(finishBtn, SIGNAL(released() ), this, SLOT(handleFinishBtnSlot() ) );
    connect(helpBtn, SIGNAL(released() ), this, SLOT(handleHelpBntSlot() ) );
    connect(cancelBtn, SIGNAL(released() ), this, SLOT(handleCancelBtnSlot() ) );
    
    outerLayout->addWidget(newProjectPage_1Ptr, 0, 0);
    outerLayout->addLayout(buttonLayout, 1, 0, Qt::AlignBottom);
}


void NewProjectGui::swapBackPage()
{
    switch(currentPage)
    {
        case PAGE_TWO:
            unloadPage_2();
            loadPage_1();
            currentPage = PAGE_ONE;
            backBtn->setEnabled(false);
            break;
        case PAGE_THREE:
            unloadPage_3();
            loadPage_2();
            currentPage = PAGE_TWO;
            break;
        case PAGE_FOUR:
            unloadPage_4();
            loadPage_3();
            currentPage = PAGE_THREE;
            nextBtn->setEnabled(true);
            finishBtn->setEnabled(false);
        default:
            cerr << "ERROR in switch at: NewProjectGui::swapBackPage()" << endl;
    }
}


void NewProjectGui::swapNextPage()
{
    switch(currentPage)
    {
        case PAGE_ONE:
            unloadPage_1();
            loadPage_2();
            currentPage = PAGE_TWO;
            backBtn->setEnabled(true);
            break;
        case PAGE_TWO:
            unloadPage_2();
            loadPage_3();
            currentPage = PAGE_THREE;
            break;
        case PAGE_THREE:
            unloadPage_3();
            loadPage_4();
            currentPage = PAGE_FOUR;
            nextBtn->setEnabled(false);
            finishBtn->setEnabled(true);
        default:
            cerr << "ERROR in switch at: NewProjectGui::swapNextPage()" << endl;
    }
}


void NewProjectGui::loadPage_1()
{
    outerLayout->addWidget(newProjectPage_1Ptr, 0, 0);
    newProjectPage_1Ptr->setVisible(true);
    newProjectPage_1Ptr->setEnabled(true);
}


void NewProjectGui::unloadPage_1()
{
    outerLayout->removeWidget(newProjectPage_1Ptr);
    newProjectPage_1Ptr->setVisible(false);
    newProjectPage_1Ptr->setEnabled(false);
}


void NewProjectGui::loadPage_2()
{
    outerLayout->addWidget(newProjectPage_2Ptr, 0, 0);
    newProjectPage_2Ptr->setVisible(true);
    newProjectPage_2Ptr->setEnabled(true);
}


void NewProjectGui::unloadPage_2()
{
    outerLayout->removeWidget(newProjectPage_2Ptr);
    newProjectPage_2Ptr->setVisible(false);
    newProjectPage_2Ptr->setEnabled(false);
}


void NewProjectGui::loadPage_3()
{
    outerLayout->addWidget(newProjectPage_3Ptr, 0, 0);
    newProjectPage_3Ptr->setVisible(true);
    newProjectPage_3Ptr->setEnabled(true);
}


void NewProjectGui::unloadPage_3()
{
    outerLayout->removeWidget(newProjectPage_3Ptr);
    newProjectPage_3Ptr->setVisible(false);
    newProjectPage_3Ptr->setEnabled(false);
}


void NewProjectGui::loadPage_4()
{
    outerLayout->addWidget(newProjectPage_4Ptr, 0, 0);
    newProjectPage_4Ptr->setVisible(true);
    newProjectPage_4Ptr->setEnabled(true);
}


void NewProjectGui::unloadPage_4()
{
    outerLayout->removeWidget(newProjectPage_4Ptr);
    newProjectPage_4Ptr->setVisible(false);
    newProjectPage_4Ptr->setEnabled(false);
}


void NewProjectGui::genRideProjDepends(QProcess* process, QString* projectRootStrPtr)
{
    process->waitForFinished();
    process->waitForFinished();
    QProcess* proc = new QProcess();

    proc->setWorkingDirectory(*projectRootStrPtr);
    QDir::setCurrent(*projectRootStrPtr);
    cout << "cCCreateing at: " << projectRootStrPtr->toStdString() << endl;
    
    //----------------
    //proc->start("/home/james/NetBeansProjects/ride/resources/setup.bash");
    QDir tmpdir(*projectRootStrPtr);
    tmpdir.cdUp();
    tmpdir.cdUp();
    QString tmpstr(tmpdir.absolutePath());
    RFile* tmp = new RFile("/tmp/setup.bash"); // !!!??? will this be automagically deleted ???!!!
    tmp->openRdWrFile();
    tmp->write("#!/bin/bash\nsource ");
    QByteArray tmpba;tmpba.append(tmpstr);
    tmp->write(tmpba);
    tmp->write("/devel/setup.bash\necho 'finished'");
    tmp->close();
    
    QStringList tmpStrLst; tmpStrLst.push_back("+x"); tmpStrLst.push_back("/tmp/setup.bash");
    proc->execute("chmod", tmpStrLst);
    proc->waitForFinished();
    proc->execute("/tmp/setup.bash");
    //tmp->remove();
    //-----------------
    
    proc->setWorkingDirectory(*projectRootStrPtr);
    QDir::setCurrent(*projectRootStrPtr);
    process->startDetached("mkdir", *(new QStringList("./.rideProject")) );
    //process->waitForFinished();
    process->execute("mkdir", *(new QStringList("./.rideProject/resources")) );
    process->execute("mkdir", *(new QStringList("./.rideProject/resources/project")) );
    process->execute("touch", *(new QStringList("./.rideProject/resources/project/configurations.xml")) );
    process->execute("mkdir", *(new QStringList("./.rideProject/resources/project/scripts")) );
    process->execute("mkdir", *(new QStringList("./.rideProject/resources/project/scripts/create")) );
    process->execute("mkdir", *(new QStringList("./.rideProject/resources/project/scripts/build")) );
    process->execute("mkdir", *(new QStringList("./.rideProject/resources/project/scripts/execute")) );
    
    //create build file
    RFile* buildFile = new RFile("./.rideProject/resources/project/scripts/build/build.bash");
    buildFile->openRdWrFile();
    buildFile->write("############################");
    buildFile->write("\n# AUTO-GENERATED RIDE FILE #");
    buildFile->write("\n############################");
    buildFile->write("\n\n#!/bin/bash");
    buildFile->write("\n\ncatkin_make $1 $2");
    buildFile->closeFile();
    
    setAbsPathToProjBuildFileStrPtr(buildFile->getAbsFilePathStrPtr() );
    
    //instead. use tmpstrlst->clear();
    tmpStrLst.clear(); tmpStrLst.push_back("+x"); tmpStrLst.push_back("./.rideProject/resources/project/scripts/build/build.bash");
    process->execute("chmod", tmpStrLst);
    
    //create rosrun file
    RFile* runFile = new RFile("./.rideProject/resources/project/scripts/execute/run.bash");
    runFile->openRdWrFile();
    runFile->write("############################");
    runFile->write("\n# AUTO-GENERATED RIDE FILE #");
    runFile->write("\n############################");
    runFile->write("\n\n#!/bin/bash");
    runFile->write("\n\nrosrun $1 $2");
    runFile->closeFile();
    
    setAbsPathToProjRunFileStrPtr(runFile->getAbsFilePathStrPtr() );
    
    tmpStrLst.clear(); tmpStrLst.push_back("+x"); tmpStrLst.push_back("./.rideProject/resources/project/scripts/execute/run.bash");
    process->execute("chmod", tmpStrLst);
    
    //create roslaunch file
    RFile* launchFile = new RFile();
    launchFile->openRdWrFile();
    launchFile->write("############################");
    launchFile->write("\n# AUTO-GENERATED RIDE FILE #");
    launchFile->write("\n############################");
    launchFile->write("\n\n#!/bin/bash");
    launchFile->write("\n\nroslaunch $1 $2");
    launchFile->closeFile();
    
    setAbsPathToProjLaunchFileStrPtr(launchFile->getAbsFilePathStrPtr() );
    
    tmpStrLst.clear(); tmpStrLst.push_back("+x"); tmpStrLst.push_back("./.rideProject/resources/project/scripts/execute/launch.bash");
    process->execute("chmod", tmpStrLst);
    
    
    
    cout << "Successfully ended file creation sequence" << endl;
    this->close();
    
    process->execute("mkdir", *(new QStringList("./.rideProject/resources/scout")) );
    process->execute("touch", *(new QStringList("./.rideProject/resources/scout/scout.qrc")) );
    process->execute("mkdir", *(new QStringList("./.rideProject/resources/scout/libraries")) );
    process->execute("mkdir", *(new QStringList("./.rideProject/resources/scout/project")) );
}


void NewProjectGui::setAbsPathToProjBuildFileStrPtr(QString* absPathToProjBuildFileStrPtr)
{
    this->absPathToProjBuildFileStrPtr = absPathToProjBuildFileStrPtr;
}


QString* NewProjectGui::getAbsPathToProjBuildFileStrPtr()
{
    return absPathToProjBuildFileStrPtr;
}


void NewProjectGui::setAbsPathToProjRunFileStrPtr(QString* absPathToProjRunFileStrPtr)
{
    this->absPathToProjRunFileStrPtr = absPathToProjRunFileStrPtr;
}


QString* NewProjectGui::getAbsPathToProjRunFileStrPtr()
{
    return absPathToProjRunFileStrPtr;
}


void NewProjectGui::setAbsPathToProjLaunchFileStrPtr(QString* absPathToProjLaunchFileStrPtr)
{
    this->absPathToProjLaunchFileStrPtr = absPathToProjLaunchFileStrPtr;
}


QString* NewProjectGui::getAbsPathToProjLaunchFileStrPtr()
{
    return absPathToProjLaunchFileStrPtr;
}


QString* NewProjectGui::toString()
{
    QString* tmp = new QString("NewProjectGui");
    tmp->append("\n- - - - - - - - - -\n");
    cout << "Fine here (0)" << endl;
    tmp->append(newProjectPage_1Ptr->toString() );
    cout << "Fine here (1)" << endl;
    tmp->append("\n");
    tmp->append(newProjectPage_2Ptr->toString() );
    cout << "Fine here (2)" << endl;
    tmp->append("\n");
    tmp->append(newProjectPage_3Ptr->toString() );
    cout << "Fine here (3)" << endl;
    tmp->append("\n");
    tmp->append(newProjectPage_4Ptr->toString() );
    cout << "Fine here (4)" << endl;
    
    return tmp;
}


NewProjectGui::~NewProjectGui()
{
    ;
}